S2VX.Game/Editor/SelectToolState.cs

```diff
@@ -1,5 +1,146 @@
-﻿namespace S2VX.Game.Editor {
+﻿using osu.Framework.Allocation;
+using osu.Framework.Extensions.Color4Extensions;
+using osu.Framework.Graphics;
+using osu.Framework.Input.Events;
+using osuTK;
+using osuTK.Graphics;
+using S2VX.Game.Story;
+using System;
+using System.Collections.Generic;
+
+namespace S2VX.Game.Editor {
     public class SelectToolState : ToolState {
+        [Resolved]
+        private S2VXStory Story { get; set; } = null;
+        [Resolved]
+        private S2VXEditor Editor { get; set; } = null;
+        private Dictionary<Note, double> SelectedNoteToTime { get; set; } = new Dictionary<Note, double>();
+        private Dictionary<Drawable, Note> NoteSelectionToNote { get; set; } = new Dictionary<Drawable, Note>();
+
+        private const float SelectionIndicatorThickness = 0.025f;
+
+        private bool DragTimelineNote { get; set; }
+        private Dictionary<Note, double> NoteToDragPointDelta { get; set; } = new Dictionary<Note, double>();
```

Comment:
I am okay with leaving these Dictionaries here for some potential future proofing for multi-selection. However, I'd like @Xenocidel opinion on it too. It's more difficult to read the code and over-complicates the current single selection code.

Reply:
I am good with this implementation. I don't think it makes sense to rewrite it just for single selection if it's working. One way to make it cleaner and avoid this complexity / usage of multiple Dictionaries is to have them be part of the Note class.